flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2341,7 +2360,7 @@ class ArrayExprLowering {
                   // Cast `e` to index type.
                   auto iv = builder.createConvert(loc, idxTy, v);
                   auto dim = sub.index();
-                  auto lb = getLBoundOrDefault(loc, exv, one, dim);
+                  auto lb = builder.readLowerBound(loc, exv, dim, one);
```

Comment:
Was moving this to the builder so it could be shared someplace else? As the builder tends to deal with the atomic particles (maybe a few smaller molecules), this seems a little misplaced in terms of the granularity. I wonder if we might want this to be a helper function instead of a method.
```c++
  Fortran::lower::readLowerBound(builder, loc, ...);
```

Reply:
That makes sense to me. I placed these here because some need to be shared and I though FirOpBuilder was the best place to do so.

I moved them to helper function, but kept them in FirOpBuilder.cpp for now (I am not sure how to call the file these kind of things should belong if we want them in a separate header/.cpp)